wrangler-core/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -0,0 +1,220 @@
+      Assert.fail("Exception in BigQuery testdata prerequisite setup " +
+              "- error in reading create table query file " + e.getMessage());
+    }
+
+    String insertDataQuery = StringUtils.EMPTY;
+    try {
+      insertDataQuery = new String(Files.readAllBytes(Paths.get(TestSetupHooks.class.getResource
+              ("/" + bqInsertDataQueryFile).toURI()))
+              , StandardCharsets.UTF_8);
+      insertDataQuery = insertDataQuery.replace("DATASET", PluginPropertyUtils.pluginProp("dataset"))
+              .replace("TABLE_NAME", bqSourceTable);
+    } catch (Exception e) {
+      BeforeActions.scenario.write("Exception in reading " + bqInsertDataQueryFile + " - " + e.getMessage());
+      Assert.fail("Exception in BigQuery testdata prerequisite setup " +
+              "- error in reading insert data query file " + e.getMessage());
+    }
+    BigQueryClient.getSoleQueryResult(createTableQuery);
+    try {
+      BigQueryClient.getSoleQueryResult(insertDataQuery);
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    PluginPropertyUtils.addPluginProp("bqSourceTable", bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " + bqSourceTable + " created successfully");
+  }
+
+  @Before(order = 1, value = "@BQ_SOURCE_TEST1")
+  public static void createTempBQTable() throws IOException, InterruptedException {
+    bqSourceTable1 = "E2E_SOURCE_" + UUID.randomUUID().toString().replaceAll("-", "_");
+    BigQueryClient.getSoleQueryResult("create table `test_automation." + bqSourceTable1 + "` " +
+            "(Customer_id INT64, First_name STRING, Last_name STRING, Age INT64, " +
+            "Address STRING, Col1 STRING, create_date DATE, update_date DATE, body STRING)");
+    try {
+      BigQueryClient.getSoleQueryResult("INSERT INTO `test_automation." + bqSourceTable1 + "` " +
+              "(Customer_id, First_name, Last_name, Age, Address, Col1, create_date, update_date, body)" +
+              " VALUES (1, 'Shelby','Shelbylastname', 20, 'Address1', 'testdata', '2021-01-28', " +
+              "'2021-01-30', 'Test' )");
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+    BeforeActions.scenario.write("BQ Source Table " + bqSourceTable1 + " created successfully");
+  }
+
+  @After(order = 1, value = "@BQ_SOURCE_TEST1")
+  public static void deleteTempSourceBQTable1() throws IOException, InterruptedException {
+    BigQueryClient.dropBqQuery(bqSourceTable1);
+    PluginPropertyUtils.removePluginProp("bqSourceTable1");
+    BeforeActions.scenario.write("BQ source Table " + bqSourceTable1 + " deleted successfully");
+    bqSourceTable1 = StringUtils.EMPTY;
+  }
+
+
+
+  @Before(order = 1, value = "@GCS_SINK_TEST1")
```

Comment:
Why are you creating so many hooks for GCS I can see that these hooks are only creating the random name for buckets this can be merged in  a single method. Which can be use to create bucket.

Reply:
resolved ...Removed all the hooks for GCS for now